### DISCUSSION TITLE
val loss in training log

### DISCUSSION THREAD

**Discussion by knifofia** [open-mmlab/mmaction2#2521] [Q&A] [answered]
Hi, folks!

I want to draw the loss function of the validation throughout the learning of my model. 
The issue is that the loss isn't print during the validation process as it is for the train (exemple below).
```
2023/05/25 11:08:04 - mmengine - INFO - Epoch(val) [2][28/28]    acc/top1: 0.3636  acc/top5: 0.9818  acc/mean1: 0.1667  data_time: 0.0052  time: 0.1863
```
```
2023/05/25 11:09:07 - mmengine - INFO - Epoch(train)      [3][100/480]  lr: 9.5381e-02  eta: 3745 days, 0:17:07  time: 0.6387  data_time: 0.0064  **loss: 1.6319**  top1_acc: 0.0000  top5_acc: 1.0000  loss_cls: 1.6319
```
Do you know a way to add it ?

I already looked to the [customize_logging doc](https://github.com/open-mmlab/mmaction2/blob/main/docs/en/advanced_guides/customize_logging.md) but didn't find it usefull for it.

I also have look at the way the logger is defined in [the mmengine repo](
https://github.com/open-mmlab/mmengine/blob/19aa1eb7800888425ce2a24d5f93b58304821821/mmengine/runner/log_processor.py#L13)

Any help is welcome, thank you in advance!

**Comment by cir7** [comment, marked as answer]
Getting the loss for val loop is a kind of `metric`, you could customize a loss metric following this [tutorial](https://mmengine.readthedocs.io/en/latest/tutorials/evaluation.html#customizing-evaluation-metrics), and add it to your config.  
We would really appreciate it if you would contribute the feature to MMAction2.
